packages/@angular/cli/plugins/bundle-budget.ts

```diff
@@ -43,54 +43,40 @@ export class BundleBudgetPlugin {
-            if (budgetCheck.thresholds.maximumWarning < size.size) {
-              compilation.warnings.push(`budgets, maximum exceeded for ${size.label}.`);
-            }
-          }
-          if (budgetCheck.thresholds.maximumError) {
-            if (budgetCheck.thresholds.maximumError < size.size) {
-              compilation.errors.push(`budgets, maximum exceeded for ${size.label}.`);
-            }
-          }
-          if (budgetCheck.thresholds.minimumWarning) {
-            if (budgetCheck.thresholds.minimumWarning > size.size) {
-              compilation.warnings.push(`budgets, minimum exceeded for ${size.label}.`);
-            }
-          }
-          if (budgetCheck.thresholds.minimumError) {
-            if (budgetCheck.thresholds.minimumError > size.size) {
-              compilation.errors.push(`budgets, minimum exceeded for ${size.label}.`);
-            }
-          }
-          if (budgetCheck.thresholds.warningLow) {
-            if (budgetCheck.thresholds.warningLow > size.size) {
-              compilation.warnings.push(`budgets, minimum exceeded for ${size.label}.`);
-            }
-          }
-          if (budgetCheck.thresholds.warningHigh) {
-            if (budgetCheck.thresholds.warningHigh < size.size) {
-              compilation.warnings.push(`budgets, maximum exceeded for ${size.label}.`);
-            }
-          }
-          if (budgetCheck.thresholds.errorLow) {
-            if (budgetCheck.thresholds.errorLow > size.size) {
-              compilation.errors.push(`budgets, minimum exceeded for ${size.label}.`);
-            }
-          }
-          if (budgetCheck.thresholds.errorHigh) {
-            if (budgetCheck.thresholds.errorHigh < size.size) {
-              compilation.errors.push(`budgets, maximum exceeded for ${size.label}.`);
-            }
-          }
+          this.checkMaximum(budgetCheck.thresholds.maximumWarning, size, compilation.warnings);
+          this.checkMaximum(budgetCheck.thresholds.maximumError, size, compilation.errors);
+          this.checkMinimum(budgetCheck.thresholds.minimumWarning, size, compilation.warnings);
+          this.checkMinimum(budgetCheck.thresholds.minimumError, size, compilation.errors);
+          this.checkMinimum(budgetCheck.thresholds.warningLow, size, compilation.warnings);
+          this.checkMaximum(budgetCheck.thresholds.warningHigh, size, compilation.warnings);
+          this.checkMinimum(budgetCheck.thresholds.errorLow, size, compilation.errors);
+          this.checkMaximum(budgetCheck.thresholds.errorHigh, size, compilation.errors);
         });
 
       });
       cb();
     });
   }
 
-  private calcualte(budget: Budget): Thresholds {
+  private checkMinimum(threshold: number, size: Size, messages: any) {
+    if (threshold) {
+      if (threshold > size.size) {
+        messages.push(`budgets, minimum exceeded for ${size.label}. `
+        + `Expected ${threshold} but was ${size.size}.`);
```

Comment:
"Expected" reads a lot like a test, which seems a little bit off for me, plus I think that units would be nice here...

Thoughts on changing the text to something like this:

```
`Budget ${threshold} was exceeded by ${size.size - threshold}.`
```

Also, putting units on these values would be helpful see [here](https://github.com/angular/angular-cli/blob/a40e801a954fd4b79e8f042f9e1692971feadf9e/packages/%40angular/cli/utilities/stats.ts#L10-L19).

Reply:
:+1: I did not dare to make `formatSize` public :)